### ISSUE TITLE
Snap on Raspberry Pi (Raspbian), 500 error page, related to `libarmmem-${PLATFORM}.so` and `/etc/ld.so.preload`

### ISSUE THREAD

**Issue by Flagelmann** [go-gitea/gitea#17462]
**Gitea Version**

All snap Gitea on Raspbian

**Description**

After installation, Gitea seems working, however, when git clone/push or view from web, Gitea reports 500 error.

The error in logs is:

```
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored
```

----

There is a workaround in comments.

**Comment by Gusted** [comment]
> So, just as a try, I just commented that single line in **/etc/ld.so.preload**: now the migration/cloning/creation repo processes works as expected (and blazing fast). Checked these processes on gitea.log in **tail -f** mode and no errors as output during the execution.

Is this a gitea-related issue then? Because every binary that would try to use that will have the same error(because $PLATFORM isn't defined). It seems to be a issue out-of-scope.

**Comment by wxiaoguang** [comment]
I high suspect it's a snap problem or snap packaging problem.

The `ld.so` is used for Linux dynamic linked programs to load other `so` files, it's a low level mechanism, not directly related to Gitea.

**Comment by Gusted** [comment]
> not directly related to Gitea.

Should we keep this issue open then? A quick search seems to indeed show that other projects has similar issues to this and that they really cannot do anything about it.

**Comment by Flagelmann** [comment]
Yep, the issue is not related directly to Gitea (at the beginning of this thread I was thinking it was related to it, but later I realized that it was related, in my case, to the following .so loaded into /etc/ld.so.preload:

/usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so

I have just commented it and restarted gitea in snap. No more error 500 and mirroring issues.

**Comment by wxiaoguang** [comment]
We should keep this issue open, there were at least 3 or 4 users reported the same problem.
This issue is the only one contains the workaround solution. I have added a `workaround` label for this issue.

And I think there might be something can be done from Gitea side:
1. Is the snap packaging correct? If the `PLATFORM` is passed in, can this problem be resolved?
2. Can Gitea detect this problem automatically and show a useful tip instead of showing the useless 500 page?

**Comment by Gusted** [comment]
> 1. Is the Gitea snap packaging correct? If the `PLATFORM` is passed in, can this problem be resolved?

Shouldn't snap handle that? Like I said it seems like a common problem, not just gitea: https://stackoverflow.com/questions/42443273/raspberry-pi-libarmmem-so-cannot-open-shared-object-file-error/50958615#50958615 https://github.com/nextcloud-snap/nextcloud-snap/issues/1438 so not sure why we should fix their problems.



> 2\. Can Gitea detect this problem automatically and show a useful tip instead of showing the useless 500 page?

That really depends on the stacktrace, but I don't think this is a catchable error within go(or any other programming language for that matter).

**Comment by wxiaoguang** [comment]
> > 1. Is the Gitea snap packaging correct? If the `PLATFORM` is passed in, can this problem be resolved?
> 
> Shouldn't snap handle that? Like I said it seems like a common problem, not just gitea: https://stackoverflow.com/questions/42443273/raspberry-pi-libarmmem-so-cannot-open-shared-object-file-error/50958615#50958615 [nextcloud-snap/nextcloud-snap#1438](https://github.com/nextcloud-snap/nextcloud-snap/issues/1438) so not sure why we should fix their problems.

It is a general issue, however, Gitea users will treat it as Gitea bug without knowledge. There are not only one issue reports before (and from now on), #19313, #17369, etc

> > 2. Can Gitea detect this problem automatically and show a useful tip instead of showing the useless 500 page?
> 
> That really depends on the stacktrace, but I don't think this is a catchable error within go(or any other programming language for that matter).

This error is reported when executing `git` cli, a simple run & check during Gitea initialization can detect such problem easily.

**Comment by Gusted** [comment]
Note: https://forum.snapcraft.io/t/snapd-on-raspbian-stretch/1419 this is only a problem due to Raspbian having incorrect files.
